var/spack/repos/builtin/packages/hiop/package.py

```diff
@@ -232,42 +230,54 @@ def cmake_args(self):
 
         return args
 
+    def _run_test_example(self, test_exe, options):
+        """Run the provided test executable with the provided options."""
+
+        exe_path = os.path.join(self.prefix.bin, test_exe)
+        if not os.path.exists(exe_path):
+            raise SkipTest("{0} is not installed".format(test_exe))
+
+        exe = which(exe_path)
+        exe(*options)
+
     # If testing on a cluster without access to home directory in a job, you may
     # set the following environment variables to prevent related errors:
     #
     # export SPACK_USER_CACHE_PATH=/tmp/spack
     # export SPACK_DISABLE_LOCAL_CONFIG=true
-    def test(self):
-        if not self.spec.satisfies("@develop") or not os.path.isdir(self.prefix.bin):
-            tty.info("Skipping: checks not installed in bin for v{0}".format(self.version))
-            return
+    def test_nlpmdsex1(self):
+        """run NlpMDsEx1.exe selfchecks"""
+        # TODO: These tests should build and run against the installed software.
+        # TODO: Set `test_requires_compiler = True` to pick up them and cmake.
+        # TODO: These tests should *never* run in the installation prefix.
```

Comment:
Are these still TODO?

Reply:
That's a great question.  Is the example built against the library and headers under `self.prefix`?  Or is it built then relocated to `self.prefix.bin`?

Reply:
Apologies for the delay. These tests are built against the shared/static library that is present in the `install/lib` directory (). I assume this is `self.prefix.lib`? The executable is installed into the `install/bin` directory, which I assume is `self.prefix.bin` (https://github.com/LLNL/hiop/blob/4b231f71dd7aa7994d841b2fa0b52ea60317f6d0/src/Drivers/MDS/CMakeLists.txt#L3).

Reply:
The package's `self.prefix` is, by default, in a subdirectory under `$SPACK_ROOT/opt/spack/architecture>/<compiler>/<package-version-hash>` but can be set in the configuration with `config:install_tree:root` (see https://spack.readthedocs.io/en/latest/config_yaml.html#config-yaml and https://spack.readthedocs.io/en/latest/config_yaml.html#install-tree-root).  AFAICT. The example is built at install time.  

What I am suggesting is to have a program that can be built and run at stand-alone test time (so days to weeks after the installation).

There's a very simple example to illustrate in the docs at https://spack.readthedocs.io/en/latest/packaging_guide.html#enabling-test-compilation.

Reply:
I think we only need to worry about this when building shared objects? I am still developing my understanding here.

I am guessing that static objects have everything that they need to run, whereas shared objects will need to have the CXX compiler available on the path?